### PR TITLE
Create unit tests for analyze_unsteady_trim

### PR DESCRIPTION
**Pull request by 20086080**
# Description

Add unit tests for analyze_unsteady_trim in trim.py covering parameter validation and operating point bounds validation.

These tests improve confidence in the trim analysis interface by verifying that invalid inputs are correctly rejected before optimization or solver execution begins.

## Motivation

analyze_unsteady_trim contains extensive validation logic for user-supplied parameters and operating point constraints. These checks are critical because they prevent invalid optimization searches and ensure that trim analysis is only performed on well-defined UnsteadyProblem configurations.

The function was previously flagged with # TEST comments but did not have dedicated unit test coverage. Adding focused validation tests makes future refactoring safer and helps ensure that documented parameter requirements remain enforced.

## Relevant Issues

Resolves unit testing component of #137.

## Changes

Added dedicated unit tests for analyze_unsteady_trim in test_unsteady_trim.py.

The following validation behavior is now tested:

1. test_problem_validation - Invalid problem parameter type raises TypeError.

2. test_multiple_airplane_movements_validation - Multiple AirplaneMovement objects raise ValueError.

3. test_boundsVCg__E_validation - Test boundsVCg__E parameter validation:

- Invalid type
- Incorrect tuple length
- Non-numeric values
- Descending bounds
- Non-positive values

4. test_alpha_bounds_validation - Test alpha_bounds parameter validation:

- Invalid type
- Incorrect tuple length
- Non-numeric values
- Descending bounds

5. test_beta_bounds_validation - Test beta_bounds parameter validation:

- Invalid type
- Incorrect tuple length
- Non-numeric values
- Descending bounds

6. test_boundsExternalFX_W_validation - Test boundsExternalFX_W parameter validation:

- Invalid type
- Incorrect tuple length
- Non-numeric values
- Descending bounds

7. test_objective_cut_off_validation - Test objective_cut_off parameter validation.

8. test_num_calls_validation - Test num_calls parameter validation.

9. test_base_operating_point_parameter_bounds_validation - Test that the base operating point values must lie within the supplied boundss:

- vCg__E
- alpha
- beta
- externalFX_W

   must lie within the user-supplied search bounds.

Testing Methodology

Tests use existing fixtures from tests.unit.fixtures.problem_fixtures to construct valid UnsteadyProblem instances.

The testing approach focuses on:

- Parameter validation
- Boundary condition validation
- Verification of documented input requirements
- Error handling through assertRaises

The tests intentionally avoid executing the optimization routines or aerodynamic solver, allowing validation logic to be tested in isolation as true unit tests.

## Dependency Updates
None

## Change Magnitude

**Minor**: Small change such as a bug fix, small enhancement, or documentation update.

## Checklist (check each item when completed or not applicable)

* [x] I am familiar with the current [contribution guidelines](https://github.com/camUrban/PteraSoftware/blob/main/CONTRIBUTING.md).
* [x] PR description links all relevant issues and follows this template.
* [x] My branch is based on `main` and is up to date with the upstream `main` branch.
* [x] All calculations use S.I. units.
* [x] Code is formatted with [black](https://github.com/psf/black) (line length = 88).
* [x] Code is well documented with block comments where appropriate.
* [x] Any external code, algorithms, or equations used have been cited in comments or docstrings.
* [x] All new modules, classes, functions, and methods have docstrings in [reStructuredText format](https://realpython.com/documenting-python-code/), and are formatted using [docformatter](https://github.com/PyCQA/docformatter) (`--in-place --black`). See the [style guide for type hints and docstrings](https://github.com/camUrban/PteraSoftware/blob/main/docs/TYPE_HINT_AND_DOCSTRING_STYLE.md) for more details.
* [x] All new classes, functions, and methods in the `pterasoftware` package use type hints. See the [style guide for type hints and docstrings](https://github.com/camUrban/PteraSoftware/blob/main/docs/TYPE_HINT_AND_DOCSTRING_STYLE.md) for more details.
* [x] If any major functionality was added or significantly changed, I have added or updated tests in the `tests` package.
* [x] Code locally passes all tests in the `tests` package.
* [x] This PR passes the ReadTheDocs build check (this runs automatically with the other workflows).
* [x] This PR passes the `ascii-only`, `black`, `codespell`, `docformatter`, `isort`, and `pre-commit-hooks` GitHub actions.
* [x] This PR passes the `mypy` GitHub action.
* [x] This PR passes all the `tests` GitHub actions.

### CODE REVIEW

File: .idea/codeStyles/codeStyleConfig.xml
Comment (camUrban) [inline .idea/codeStyles/codeStyleConfig.xml:1]: Don't delete.

File: .idea/codeStyles/Project.xml
Comment (camUrban) [inline .idea/codeStyles/Project.xml:1]: Don't delete.

File: .idea/dictionaries/project.xml
Comment (camUrban) [inline .idea/dictionaries/project.xml:1]: Don't delete.

File: .idea/inspectionProfiles/profiles_settings.xml
Comment (camUrban) [inline .idea/inspectionProfiles/profiles_settings.xml:1]: Don't delete.

File: .idea/inspectionProfiles/Ptera_Software_Default.xml
Comment (camUrban) [inline .idea/inspectionProfiles/Ptera_Software_Default.xml:1]: Don't delete.

File: .idea/scopes/to_inspect.xml
Comment (camUrban) [inline .idea/scopes/to_inspect.xml:1]: Don't delete.

File: .idea/.gitignore
Comment (camUrban) [inline .idea/.gitignore:1]: Don't delete.

File: .idea/.name
Comment (camUrban) [inline .idea/.name:1]: Don't delete.

File: .idea/copilot.data.migration.agent.xml
Comment (camUrban) [inline .idea/copilot.data.migration.agent.xml:1]: Don't delete.

File: .idea/copilot.data.migration.ask.xml
Comment (camUrban) [inline .idea/copilot.data.migration.ask.xml:1]: Don't delete.

File: .idea/copilot.data.migration.ask2agent.xml
Comment (camUrban) [inline .idea/copilot.data.migration.ask2agent.xml:1]: Don't delete.

File: .idea/copilot.data.migration.edit.xml
Comment (camUrban) [inline .idea/copilot.data.migration.edit.xml:1]: Don't delete.

File: .idea/icon.svg
Comment (camUrban) [inline .idea/icon.svg:1]: Don't delete.

File: .idea/jsonCatalog.xml
Comment (camUrban) [inline .idea/jsonCatalog.xml:1]: Don't delete.

File: .idea/markdown.xml
Comment (camUrban) [inline .idea/markdown.xml:1]: Don't delete.

File: .idea/misc.xml
Comment (camUrban) [inline .idea/misc.xml:1]: Don't delete.

File: .idea/modules.xml
Comment (camUrban) [inline .idea/modules.xml:1]: Don't delete.

File: .idea/other.xml
Comment (camUrban) [inline .idea/other.xml:1]: Don't delete.

File: .idea/PteraSoftware.iml
Comment (camUrban) [inline .idea/PteraSoftware.iml:1]: Don't delete.

File: .idea/statistic.xml
Comment (camUrban) [inline .idea/statistic.xml:1]: Don't delete.

File: .idea/vcs.xml
Comment (camUrban) [inline .idea/vcs.xml:1]: Don't delete.

File: .idea/watcherTasks.xml
Comment (camUrban) [inline .idea/watcherTasks.xml:1]: Don't delete.

File: tests/unit/test_unsteady_trim.py [hunk 1]
```diff
@@ -0,0 +1,294 @@
+import unittest
```
Comment (camUrban) [inline tests/unit/test_unsteady_trim.py:3]: Please add a module docstring (see other unit test files for the proper format). Also, you'll want to add this file's docstring to the list of docstrings in tests/unit/__init__.py and also import it there (in alphabetical order).

File: tests/unit/test_unsteady_trim.py [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+        with self.assertRaises(ValueError):
+            ps.trim.analyze_unsteady_trim(
+                problem=self.problem,
+                boundsVCg__E=(1.0, 100.0),
+                alpha_bounds=(-20.0, 20.0),
+                beta_bounds=(-20.0, 20.0),
+                boundsExternalFX_W=(10.0, -10.0),
+            )
+
+    def test_objective_cut_off_validation(self):
```
Comment (camUrban) [inline tests/unit/test_unsteady_trim.py:208]: It might be good to also add an assertRaises test on negative objective_cut_off values as well.

File: tests/unit/test_unsteady_trim.py [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+            ps.trim.analyze_unsteady_trim(
+                problem=self.problem,
+                boundsVCg__E=(1.0, 100.0),
+                alpha_bounds=(-20.0, 20.0),
+                beta_bounds=(-20.0, 20.0),
+                boundsExternalFX_W=(-1000.0, 1000.0),
+                objective_cut_off=0.0,
+            )
+
+    def test_num_calls_validation(self):
```
Comment (camUrban) [inline tests/unit/test_unsteady_trim.py:231]: For num_calls, we also want to throw out non-integer values and negative values.

File: .gitignore
Comment (camUrban) [inline .gitignore:1]: No need to add anything to this file.

**Review by camUrban — CHANGES_REQUESTED** [review]
Hi  @20086080, thanks for the work here. Your tests look great! I left a few suggestions for those, but the bigger thing is all the `.idea` file changes. We still don't want those modified or deleted or added to `.gitignore`. I think the best way to fix it is to follow the steps 1-6 from [my comment here](https://github.com/camUrban/PteraSoftware/pull/169#issuecomment-4401497259). The only thing to do different this time is to include the file names of all the `.idea` files and also `.gitignore` for step 2.

It seems like the steps I provided last time didn't quite work in getting your git install to ignore these. Therefore, in the future, I'd recommend not using `git add .` before committing. Instead, you can `git add <file_1> <file_2> ...` with the files you actually want to change, which should prevent your next `git commit -m "<some message>"` from picking up any extraneous changes. 😊

**Comment by 20086080** [comment]
Hi @camUrban 

I have added in the 2 test changes and doc string. 
Unfortunately I am still a bit stuck with the .idea/ files. Am on Step 4 of https://github.com/camUrban/PteraSoftware/pull/169#issuecomment-4401497259. Step 1 - 3 were fine (for Step 2 I individually added in the files in the .git/info/exclude). 
For step 4 - This is what my git status is showing me. Please let me know what best to do. The changes in the test file were committed before I started trying to fix the .idea/ files. 
 
Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
        new file:   .idea/.gitignore
        new file:   .idea/.name
        new file:   .idea/PteraSoftware.iml
        new file:   .idea/codeStyles/Project.xml
        new file:   .idea/codeStyles/codeStyleConfig.xml
        new file:   .idea/copilot.data.migration.agent.xml
        new file:   .idea/copilot.data.migration.ask.xml
        new file:   .idea/copilot.data.migration.ask2agent.xml
        new file:   .idea/copilot.data.migration.edit.xml
        new file:   .idea/dictionaries/project.xml
        new file:   .idea/icon.svg
        new file:   .idea/inspectionProfiles/Ptera_Software_Default.xml
        new file:   .idea/inspectionProfiles/profiles_settings.xml
        new file:   .idea/jsonCatalog.xml
        new file:   .idea/markdown.xml
        new file:   .idea/misc.xml
        new file:   .idea/modules.xml
        new file:   .idea/other.xml
        new file:   .idea/scopes/to_inspect.xml
        new file:   .idea/statistic.xml
        new file:   .idea/vcs.xml
        new file:   .idea/watcherTasks.xml

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   .idea/dictionaries/project.xml

**Comment by camUrban** [comment]
Hi @20086080, thanks for reaching back out. Don't worry, you are mostly there! One quick question to be sure we're on the same page:

When you ran step 1 from [the comment](https://github.com/camUrban/PteraSoftware/pull/169#issuecomment-4401497259), did you run it verbatim, or did you substitute the old branch name with the current branch? You can check by looking at the top few lines in your `git status` output. They should say what branch you are on. We want to be on `Issue-137`.

Once you're sure you are on the right branch, the next step is to also add the `.gitignore` file to the changes you'll revert. To do that, run `git checkout HEAD~1 -- .gitignore`. Now, the "Changes to be committed list: in `git status` should look something like:

```
modified file: .gitignore
new file: .idea/.gitignore
new file: .idea/.name
new file: .idea/PteraSoftware.iml
new file: .idea/codeStyles/Project.xml
new file: .idea/codeStyles/codeStyleConfig.xml
new file: .idea/copilot.data.migration.agent.xml
new file: .idea/copilot.data.migration.ask.xml
new file: .idea/copilot.data.migration.ask2agent.xml
new file: .idea/copilot.data.migration.edit.xml
new file: .idea/dictionaries/project.xml
new file: .idea/icon.svg
new file: .idea/inspectionProfiles/Ptera_Software_Default.xml
new file: .idea/inspectionProfiles/profiles_settings.xml
new file: .idea/jsonCatalog.xml
new file: .idea/markdown.xml
new file: .idea/misc.xml
new file: .idea/modules.xml
new file: .idea/other.xml
new file: .idea/scopes/to_inspect.xml
new file: .idea/statistic.xml
new file: .idea/vcs.xml
new file: .idea/watcherTasks.xml
```

Once you've done that, you can just run a normal commit and push:

```
git commit -m "Restore .gitignore and .idea"
```
```
git push
```

After that, let me know if you have any other issues walking through step 8!

**Comment by camUrban** [comment]
Hey @20086080! I ended up implementing those tidying chores and small test additions myself. This was just because `main` had changed a bit since you last committed here, so I needed to do a manual merge to update the branch anyway. I know you said you'd already implemented some of these tidying tasks yourself, and were just stuck on a few of them, so I added you as a co-author on the commit.

Thanks again for the new unit tests! Another reason I finished this up myself was because I'm rushing to get a new release prepped, and I wanted to include your work 😄

**Comment by 20086080** [comment]
@camUrban Thank you so much :-).
